GVRf/Framework/src/org/gearvrf/GVRSceneObject.java

```diff
@@ -239,15 +239,18 @@ public String getName() {
     }
 
     /**
-     * Set the (optional) name of the object.
+     * Set the (optional) name of the object. Scene object names are not needed:
+     * they are only for the application's convenience.
      * 
-     * Scene object names are not needed: they are only for the application's
-     * convenience.
-     * 
-     * @param name
-     *            Name of the object.
+     * @param name Name of the object.
+     * @throws IllegalArgumentException if input parameter is null or empty.
      */
     public void setName(String name) {
+        if (name == null || name.trim().isEmpty()) {
```

Comment:
No,  with map, I didn’t mean anything related to <SceneObject>*. I was saying possible uses of maps in the applications themselves, where they will use the name as a key with logical names.